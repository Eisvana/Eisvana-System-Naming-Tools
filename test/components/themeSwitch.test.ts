import ThemeSwitch from '@/components/ThemeSwitch.vue';
import { beforeEach, describe, expect, it } from 'vitest';
import { mount, type VueWrapper } from '@vue/test-utils';

describe('ThemeSwitch', () => {
  let wrapper: VueWrapper;

  beforeEach(() => {
    wrapper = mount(ThemeSwitch);
  });

  it('should render correctly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should swap theme if clicked', async () => {
    const currentTheme = document.documentElement.dataset.theme;
    const newTheme = currentTheme === 'dark' ? 'light' : 'dark';
    const button = wrapper.find('button');
    await button.trigger('click');

    expect(document.documentElement.dataset.theme).toBe(newTheme);
  });
});
