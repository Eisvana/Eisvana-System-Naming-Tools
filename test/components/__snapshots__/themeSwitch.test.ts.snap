// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ThemeSwitch > should render correctly 1`] = `"<button data-v-7e4fbe6f="" class="themeswitcher" id="themeSwitch"> Switch Theme </button>"`;
